### PR TITLE
Add 9.0 Blazor Web App using Microsoft.Idenity.Web and Entra

### PR DESCRIPTION
**Pull request by halter73**

### CODE REVIEW

File: 9.0/BlazorWebAppEntra/BlazorWebAppEntra/LoginLogoutEndpointRouteBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Microsoft.AspNetCore.Routing;
+
+internal static class LoginLogoutEndpointRouteBuilderExtensions
+{
+    internal static IEndpointConventionBuilder MapLoginAndLogout(this IEndpointRouteBuilder endpoints)
+    {
+        var group = endpoints.MapGroup("");
```
Comment (guardrex) [inline 9.0/BlazorWebAppEntra/BlazorWebAppEntra/LoginLogoutEndpointRouteBuilderExtensions.cs:12]:
```suggestion
        var group = endpoints.MapGroup(string.Empty);
```

File: 9.0/BlazorWebAppEntra/BlazorWebAppEntra/wwwroot/app.css [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+.form-floating > .form-control-plaintext::placeholder, .form-floating > .form-control::placeholder {
+    color: var(--bs-secondary-color);
+    text-align: end;
+}
+
+.form-floating > .form-control-plaintext:focus::placeholder, .form-floating > .form-control:focus::placeholder {
+    text-align: start;
+}
```
Comment (guardrex) [inline 9.0/BlazorWebAppEntra/BlazorWebAppEntra/wwwroot/app.css:60]:
```suggestion
}

```

**Review by guardrex — APPROVED** [review]
Super NITs ... I'll go ahead a place them.

**Comment by Eagle3386** [comment]
@guardrex May I ask what "Super NITs" means? 😅

**Comment by guardrex** [comment]
https://stackoverflow.com/questions/27810522/what-does-nit-mean-in-hacker-speak

**Comment by Stroniax** [comment]
I have tried to follow this guide and it worked for prompting login to access pages on the app. When I make a call to an endpoint (controller) with an authorization requirement I get back a 200 with the microsoft login page html, instead of receiving a 401 or 403 that I would expect. Is there another step I need to set up authorization for the backend? I have noticed that `app.UseAuthentication()` and `app.UseAuthorization()` are not in the `Program.cs` file of this project.

**Comment by guardrex** [comment]
Stand-by, @Stroniax ... I'll see when Halter is planning on placing that sample. It was agreed to do it offline, but he didn't say when he'd get to it.

**Comment by uwer62** [comment]
First of all I would like to thank you for the sample "BlazorWebAppEntra". I spent a lot of hours (unsuccessfully) to get this to work in Blazor 8.

I have tried to modify the sample to support the Graph API for the logged in user.

In order to achieve this I returned it to a Interactive Server Web App ( removed the client, added the client files to the server project and enabled the InterActiveServer RenderMode).

Then I had to add the follwing lines:

```
builder.Services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
    .AddMicrosoftIdentityWebApp(builder.Configuration.GetSection("AzureAd"))
==>    .EnableTokenAcquisitionToCallDownstreamApi(new string[] { "user.read" })
==>    .AddDistributedTokenCaches();

==> builder.Services.AddMicrosoftGraph(options => options.Scopes = new List<string>() { "user.read" });
```
Now I  get a functional GraphServiceClient injected and I am able to send an email via Graph API after I login within the WebApp.

My problem is this: 

If I close the website and open it again **and** the Website recognizes me as an already logged in user then the credential info and the weather-page are both working fine. The sendmail call fails with 

`(Error) IDW10502: An MsalUiRequiredException was thrown due to a challenge for the user. See https://aka.ms/ms-id-web/ca_incremental-consent.`

If I log out and log in again (within the same session) it is working.

It seems that the middleware doesn't automatically pick up the required token.

Is this a bug? Am I missing something that needs to be called in this scenario?

**Comment by guardrex** [comment]
@uwer62 ... That's a duplicate request of https://github.com/dotnet/AspNetCore.Docs/issues/33147. BTW, we don't work closed issues/PRs. In the future, please open a new docs issue using the **Open a documentation issue** link/feedback form at the bottom of the article. Use of the form adds metadata to your GitHub issue that cross-links the topic and pings the author automatically for a faster response.

**Pull request merged**